azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,175 @@
+                    "signingRules": {
+                        "extensionSigned": <true, false>
+                    }
+                }
+        }
+        """
+        self._policy_engine_enabled = False
+        self._engine = None
+        if not self.is_policy_enforcement_enabled():
+            self.log_policy(msg="Policy enforcement is not enabled.")
+            return  # policy_engine_enabled is not set to True
+
+        # If unsupported, this call will raise an error
+        self.check_policy_enforcement_supported()
+
+        try:
+            self._engine = regorus.Engine(policy_file=policy_file, rule_file=rule_file)
+            self._policy_engine_enabled = True
+        except Exception as ex:
+            msg = "Failed to initialize Regorus policy engine: '{0}'".format(ex)
+            self.log_policy(msg=msg)
+            raise PolicyError(msg)
+
+    @classmethod
+    def log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+        """
+        Log information to console and telemetry.
+        """
+        if is_success:
+            logger.info(msg)
+        else:
+            logger.error(msg)
+        if send_event:
+            add_event(op=op, message=msg, is_success=is_success)
+
+    @staticmethod
+    def is_policy_enforcement_enabled():
+        """Check whether user has opted into policy enforcement feature"""
+        # TODO: The conf flag will be removed post private preview. Before public preview, add checks
+        # according to the planned user experience (TBD).
+        return conf.get_extension_policy_enabled()
+
+    @staticmethod
+    def check_policy_enforcement_supported():
+        """
+        Check that both platform architecture and distro/version are supported.
+        If supported, do nothing.
+        If not supported, raise PolicyError with user-friendly error message.
+        """
+        osutil = get_osutil()
+        arch = osutil.get_vm_arch()
+        # TODO: surface as a user error with clear instructions for fixing
+        msg = "Attempted to enable policy enforcement, but feature is not supported on "
+        if arch not in POLICY_SUPPORTED_ARCHITECTURE:
+            msg += " architecture " + str(arch)
+        elif DISTRO_NAME not in POLICY_SUPPORTED_DISTROS_MIN_VERSIONS:
+            msg += " distro " + str(DISTRO_NAME)
+        else:
+            min_version = POLICY_SUPPORTED_DISTROS_MIN_VERSIONS.get(DISTRO_NAME)
+            if DISTRO_VERSION < min_version:
+                msg += " distro " + DISTRO_NAME + " " + DISTRO_VERSION + ". Policy is only supported on version " + \
+                        str(min_version) + " and above."
+            else:
+                return  # do nothing if platform is supported
+        raise PolicyError(msg)
+
+
+    @property
+    def policy_engine_enabled(self):
+        """This property tracks whether the feature is enabled and Regorus engine has been successfully initialized"""
```

Comment:
left-over comment: "and Regorus engine has been successfully initialized""

Reply:
removed this property